casepro/pods/base.py

```diff
@@ -0,0 +1,46 @@
+from confmodel import fields, Config as ConfmodelConfig
+from django.apps import AppConfig
+from django.conf.urls import url
+from django.http import HttpResponse
+
+
+class PodConfig(ConfmodelConfig):
+    '''
+    This is the config that all pods should use as the base for their own
+    config.
+    '''
+    index = fields.ConfigInt(
+        "A unique identifier for the specific instance of this pod."
+        "Automatically determined and set in the pod registry.",
+        required=True)
+
+
+class Pod(object):
+    '''
+    The base class for all pod plugins.
+    '''
+    config_cls = PodConfig
+    name = 'Default pod name'
+    controller = None
+    directive = None
+
+    def __init__(self, config):
+        self.config = self.config_cls(config)
+
+    @property
+    def url_patterns(self):
+        return (
+            url(r'read/%d/$' % self.config.index, self.request_callback),
```

Comment:
There needs to be a way for the client side to figure out what this URL is.

@justinvdm mentioned the idea of having two fixed URLs (that we all discussed previously) that then just map to the appropriate data pod and call a method there. I'm still leaning towards that as a better idea that allowing complex arbitrary URLs (although maybe for very complex pods the ability to add custom URLs per pod-type would still be useful). I'd be happy to leave that our entirely until we encounter the need though.